### PR TITLE
Test NMO operator upgrade from v0.17.0 to latest

### PR DESCRIPTION
**Pull request by razo7**
Perform an operator upgrade test from the last released version to the latest commit in `main`.
Upgrade an old operator to a new one on the same OCP cluster, see https://docs.ci.openshift.org/docs/how-tos/testing-operator-sdk-operators/#operator-upgrade-testing

### CODE REVIEW

**Comment by razo7** [comment]
/pj-rehearse pull-ci-medik8s-fence-agents-remediation-main-4.16-far-upgarde-latest

**Comment by razo7** [comment]
FAR doesn't have the [`bundle-run-update` target](https://github.com/medik8s/node-maintenance-operator/pull/120), so switching the change and testing for NMO ATM

**Comment by razo7** [comment]
/pj-rehearse pull-ci-medik8s-node-maintenance-operator-main-4.16-openshift-e2e

**Comment by razo7** [comment]
> * why spinning up a new cluster for this test instead of using the one we have alrady?
> * what about doing what the docs suggest, using 3 steps: install, upgrade, e2e test? ([docs.ci.openshift.org/docs/how-tos/testing-operator-sdk-operators/#operator-upgrade-testing](https://docs.ci.openshift.org/docs/how-tos/testing-operator-sdk-operators/#operator-upgrade-testing))
> 
> /hold

Good input, I have followed your suggestions, applied them for NMO, and I will now add the upgrade utility in FAR

**Comment by razo7** [comment]
/pj-rehearse pull-ci-medik8s-node-maintenance-operator-main-4.16-openshift-e2e

**Comment by razo7** [comment]
/pj-rehearse pull-ci-medik8s-node-maintenance-operator-main-4.16-openshift-e2e

**Comment by razo7** [comment]
/pj-rehearse pull-ci-medik8s-node-maintenance-operator-main-4.16-openshift-e2e

**Comment by razo7** [comment]
/pj-rehearse pull-ci-medik8s-node-maintenance-operator-main-4.16-openshift-e2e

**Review by clobrano — CHANGES_REQUESTED** [review]

File: ci-operator/config/medik8s/node-maintenance-operator/medik8s-node-maintenance-operator-main__4.16.yaml [hunk 1]
```diff
@@ -43,22 +39,52 @@ tests:
...
+      env:
+      - name: OPERATOR_NAMESPACE
+      from: src
+      resources:
+        requests:
+          cpu: 100m
+          memory: 200Mi
+    - as: e2e-upgrade-operator-latest
+      cli: latest
+      commands: BUNDLE_IMG="$OO_BUNDLE" make bundle-run-update
```
[the remark this answers is not in the document]
Comment (razo7) [reply ci-operator/config/medik8s/node-maintenance-operator/medik8s-node-maintenance-operator-main__4.16.yaml:66]: It [does](https://github.com/medik8s/node-maintenance-operator/blob/main/Makefile#L202), but OPERATOR_NAMESPACE is [defined already as an environment variable ](https://github.com/openshift/release/pull/50559/files#diff-7d3049ecc7981664e5c32a8c217f3ceb2c413bc9fd4f61bb6b5c07f81ff36dd6R44)for the main test (`openshift-e2e`)

**Review by clobrano — APPROVED** [review]

**Pull request merged**